OpenRA.Mods.Common/Traits/Render/WithChargeOverlay.cs

```diff
@@ -0,0 +1,72 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	[Desc("Render overlay that varies the animation frame based on the AttackChargestrait's charge level.")]
+	class WithChargeOverlayInfo : ITraitInfo, Requires<WithSpriteBodyInfo>, Requires<RenderSpritesInfo>
```

Comment:
In this case I *could, but that would be bad practice, so I won't.

`WithChargeOverlayInfo` doesn't derive from or directly depend on `WithSpriteBodyInfo`, so it shouldn't assume anything about the private implementation details of that class.  If somebody changed `WithSpriteBodyInfo` to not `Requires<RenderSpritesInfo>` then this trait would silently break.